React Native Docs

crypblizz8 opened the pull request:
Context: With the growing development in RN and its value, we dedicate a section specific to React Native. It has been structured to provide information on as many SDK/APIs we use for the platform.

ToDo: 
- [x] Add to top `sidebar`.
- [x] Check for mobile
- [x] Updated Examples and Resources

Structure
- [x] Has its own section
- [x] Web3Wallet

Later PR / when ready
- Web3Modal
- Web3Inbox

ignaciosantise commented on docs/reactnative/web3wallet/GettingStarted.md:
```diff
@@ -0,0 +1,27 @@
+
+[Web3Wallet SDK](https://medium.com/walletconnect/simplifying-integration-for-wallet-developers-with-the-new-web3wallet-sdk-8706b69e149c) simplifies the integration process for wallet developers by combining our Sign and Auth APIs. Only V2 WCURIs will work with this SDK as V1 is being deprecated by 28 June 2023.
+
+We have written a simple tutorial for Expo on how to get started. This is same approach is applicable for React Native CLI. At any point of development, feel free to reach out via Github Discussions or Discord in the `wallet-dev-support` channel. [ADD LINKS]
```
`This same approach is applicable `

ignaciosantise replied:
also should you change that `[ADD LINKS]`?

ignaciosantise replied:
should we also add links to Discord and GH discussions?

ignaciosantise commented on docs/reactnative/web3wallet/Usage.md:
````diff
@@ -0,0 +1,219 @@
+The `session_update` event is emitted from the wallet when the session is updated by calling `updateSession`. To update a session, pass in the `topic` and the new namespace.
+
+```javascript
+await web3wallet.updateSession({ topic, namespaces: newNs })
````
example of a topic?

crypblizz8 commented:
> Q: How to handle all the extra content the JS sidebar?
I believe JS <> RN will be separate. I removed the React-native guide from JS.

Q: Does the format differ from the other sections? Eg Installation, Usage. Why not keep it consistent?
It does not differ. Only difference I added was a opinionated `Introduction` section so it can guide them to the various SDKs. This was based off how W3M JS docs currently operate. In future with W3M + W3I, this serves as a good launchpad and direction to the right SDK. CC @chadyj

chadyj commented:
> I believe JS <> RN will be separate. I removed the React-native guide from JS.

Right, but then the RN section is missing all the other content.
Even Flutter has some of it https://docs.walletconnect.com/2.0/flutter/installation

crypblizz8 commented:
> Right, but then the RN section is missing all the other content.
> Even Flutter has some of it https://docs.walletconnect.com/2.0/flutter/installation

Those `Advanced API` sections do not seem relevant until we have W3M section up.

crypblizz8 commented:
> RN is theonly one without an "installation" section. Instead it has "getting started". Why the difference?

Thanks for the spot, have restructured!

finessevanes commented on docs/reactnative/Introduction.md:
```diff
@@ -0,0 +1,38 @@
+
+# Introduction
+
+Welcome to the React Native documentation for WalletConnect! We have guides for both Web3Wallet (wallet side) and Web3Modal (Dapp side) coming soon.
```
```suggestion
Welcome to the React Native documentation for WalletConnect! We will soon have guides available for both Web3Wallet (wallet side) and Web3Modal (Dapp side).

finessevanes commented on docs/reactnative/web3wallet/Installation.md:
```diff
@@ -0,0 +1,51 @@
+
+# Installation
+
+[Web3Wallet SDK](https://medium.com/walletconnect/simplifying-integration-for-wallet-developers-with-the-new-web3wallet-sdk-8706b69e149c) simplifies the integration process for wallet developers by combining our Sign and Auth APIs. Only V2 WCURIs will work with this SDK as V1 is being deprecated by 28 June 2023.
```
Would be cool to link WCURIs to more info :)

finessevanes commented on docs/reactnative/web3wallet/wallet-usage.md:
```diff
@@ -0,0 +1,193 @@
+
+# Wallet Usage
+
+The followings steps describe the process of creating a React Native wallet with the Web3Wallet SDK.
```
Love the steps outlined like this

finessevanes commented on docs/reactnative/web3wallet/wallet-usage.md:
```diff
@@ -0,0 +1,193 @@
+
+## Pairing, Session Approval & Rejection
+
+In order to connect with a dapp, you will need to receive a WalletConnect URI (WCURI) and this will talk to our protocol to facilitate a pairing session. Therefore, you will need a test dApp in order to communicate with the wallet. We recommend testing with our [React V2 Dapp](https://react-app.walletconnect.com/) as this is the most up-to-date development site.
```
```suggestion
In order to connect with a dapp, you will need to receive a WalletConnect URI (WCURI) and this will talk to our protocol to facilitate a pairing session. Therefore, you will need a test dapp in order to communicate with the wallet. We recommend testing with our [React V2 Dapp](https://react-app.walletconnect.com/) as this is the most up-to-date development site.
```

finessevanes replied:
we mostly have `dapp` over `dApp` in the docs

crypblizz8 commented on docs/reactnative/web3wallet/Installation.md:
```diff
@@ -0,0 +1,51 @@
+
+[Web3Wallet SDK](https://medium.com/walletconnect/simplifying-integration-for-wallet-developers-with-the-new-web3wallet-sdk-8706b69e149c) simplifies the integration process for wallet developers by combining our Sign and Auth APIs. Only V2 WCURIs will work with this SDK as V1 is being deprecated by 28 June 2023.
+
+We have written a simple tutorial for Expo on how to get started. This same approach is applicable for React Native CLI. At any point of development, feel free to reach out via [Github Discussions](https://github.com/orgs/WalletConnect/discussions) or Discord in the `wallet-dev-support` [channel.](https://discord.com/channels/492410046307631105/1040018700734038169)
```
RN Cli is via the terminal client + Expo is more a third party software for the development. Expo has become easier to build on in recent years.

I believe as an RN dev it is very clear that you know what the difference is so will leave as it is
